File: C/diagonal negativa.c
#include <stdio.h>

int main()
{
    int n, qnt_negativos = 0;

    printf("qual a ordem da matriz?: ");
    scanf("%i", &n);

    int mat[n][n];

    for (int l = 0; l<n;l++){
        for(int c = 0; c<n;c++){
            printf("\ndigite o elemento da posicao [%i, %i]: ", l+1, c+1);
            scanf("%i", &mat[l][c]);

        }
    }
    for (int l = 0; l<n;l++){
        for(int c = 0; c<n;c++){
            if (mat[l][c]<0){
                qnt_negativos +=1;
            }
        }
    }

    printf("\nquantidade de negativos: %i", qnt_negativos);
    printf("\nDiagonal principal: ");
    for (int i = 0; i < n; i++){
        printf(" %i ", mat[i][i]);
    }
    return 0;
}

File: Python/diagonal_negativa.py
N = int(input("Digite a ordem da matriz (no maximo 10): "))
while N>10:
    N = int(input("Digite a ordem da matriz (no maximo 10): "))

mat = [[0 for x in range (N)] for x in range (N)]

qnt_negativo = 0

for i in range(0, N):
    for j in range(0, N):
        mat[i][j] = int(input(f"Digite o valor do elemento[{i}, {j}]: "))
        if mat[i][j]<0:
            qnt_negativo = qnt_negativo + 1
    
print("DIAGONAL PRINCIPAL: ")
for i in range(0, N):
    for j in range(0, N):
        if i == j:
            print(f"{mat[i][j]} ", end="")
print(f"\nQUANTIDADE DE NEGATIVOS = {qnt_negativo}")